core/src/main/java/com/exadel/aem/toolkit/bundle/customlists/datasource/ChildResourcesDatasource.java

```diff
@@ -0,0 +1,201 @@
+            Integer limit = expressionHelper.get(dsCfg.get(LIMIT, String.class), Integer.class);
+            String itemResourceType = dsCfg.get(PN_RESOURCE_TYPE, String.class);
+
+            Resource parent = parentPath != null ? resolver.getResource(parentPath) : null;
+            if (parent != null) {
+                Iterator<Resource> resources = getValidChildren(resolver, parent).iterator();
+                dataSource = createDataSource(resources, offset, limit, itemResourceType);
+            }
+        }
+        request.setAttribute(DataSource.class.getName(), dataSource);
+    }
+
+    /**
+     * Creates a {@code datasource} limited by 'offset' and 'limit' parameter values
+     * from {@code Iterator<Resource>} instance
+     *
+     * @param resources {@code Iterator<Resource>} instance of valid child pages
+     * @param offset    The integer number of items that should be skipped
+     * @param limit     The integer number of items that should be included
+     * @param itemRT    Resource type of items
+     * @return {@code DataSource} object
+     */
+    private DataSource createDataSource(Iterator<Resource> resources, Integer offset, Integer limit, String itemRT) {
+        return new AbstractDataSource() {
+            @SuppressWarnings("unchecked")
+            @Override
+            public Iterator<Resource> iterator() {
+                Iterator<Resource> it = new PagingIterator<>(new FilterIterator<>(resources, o -> {
+                    String name = ((Resource) o).getName();
+                    return !name.startsWith(REP_PREFIX) && !name.equals(JCR_CONTENT);
+                }), offset, limit);
+
+                return new TransformIterator(it, o -> new ResourceWrapper((Resource) o) {
+                    @Nonnull
+                    @Override
+                    public String getResourceType() {
+                        return itemRT;
+                    }
+                });
+            }
+        };
+    }
+
+    /**
+     * Retrieves the list of item resources, which are either custom lists themselves, or folders
+     * that may contain lists inside
+     *
+     * @param resolver An instance of ResourceResolver
+     * @param parent {@code Resource} instance used as the source of markup
+     * @return a list of {@link Resource}s
+     */
+    private List<Resource> getValidChildren(ResourceResolver resolver, Resource parent) {
+        return getChildrenStream(parent)
+            .filter(resource -> checkResourceAsTemplate(resolver, resource) || checkResourceAsFolder(resource))
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Checks whether the resource is a custom list
+     *
+     * @param resolver An instance of ResourceResolver
+     * @param resource {@code Resource} instance used as the source of markup
+     * @return True or false
+     */
+    private static boolean checkResourceAsTemplate(ResourceResolver resolver, Resource resource) {
```

Comment:
I suggest simplifying the name of this method to isTemplate

Reply:
Thank you, done